3/README.md

```diff
@@ -0,0 +1,261 @@
+
+## Specification
+
+ A Tango Command is a strict definition of a function apply at the Device context:
+
+ * The Command SHALL have a unique identifier, called the Command Name
+
+ * The Command input or output are called Argument
+
+ * The Command SHALL have one input Argument, called ARGIN
+
+ * The Command SHALL have one output Argument, called ARGOUT
+ 
+ An Argument represents the input and output of a Command and its specification can only applied to a Command as below:
+
+ * Argument SHALL have an Argument Type which set the type of the Argument Value
+
+ * Argument MAY have an Argument Description which describe the meaning of the Argument Value
+  
+ An Argument Value only exists during the phase of execution of a Command. 
+ In this case a Command MAY have 2 Arguments Values, one for ARGIN and one for ARGOUT.
+ 
+ The Argument Type, Argument Value, Argument Representation and Argument Description are specified below as litteral representation. This SHOULD NOT constraint the implementation to use a binary format for the transport layer.
+
+ ``` ABNF
+ argument = dev-void | dev-double | dev-boolean | dev-float | dev-short | dev-long | dev-long64 | dev-string | dev-uchar | dev-ushort | dev-ulong | dev-ulong64 | dev-boolean-array | dev-double-array | dev-float-array | dev-short-array | dev-long-array | dev-long64-array | dev-char-array | dev-string-array | dev-ushort-array | dev-ulong-array | dev-ulong64-array | dev-long-string-array | dev-double-string-array | dev-boolean-array | dev-encoded | dev-encoded-array 
+ 
+ argument-desc = 0*CHAR
+
+ ; Float Definition inspired by python float definition 
+ ; Some examples of floating point literals: 3.14 10. .001 1e100 3.14e-10
+ floatnumber = [ "-" ] pointfloat | exponentfloat
+ pointfloat = [intpart] fraction | intpart "."
+ exponentfloat:  ( %d1-9 DIGIT* | pointfloat) exponent
+ intpart:        %d1-9 DIGIT* | %d0
+ fraction:       "." DIGIT+
+ exponent:       ("e"|"E") ["+"|"-"] DIGIT+
+
+ dev-void = dev-void-type WSP dev-void-value
+ dev-void-type = "DEVVOID" | "DevVoid"
+ dev-void-value = NULL
+ 
+ dev-double = dev-double-type WSP dev-double-value
+ dev-double-type = "DEVDOUBLE" | "DevDouble"
+ dev-double-value = floatnumber ; double-precision normalized numbers in IEC 60559
+
+ dev-boolean = dev-boolean-type WSP dev-boolean-value
+ dev-boolean-type = "DEVBOOLEAN" | "DevBoolean"
+ dev-boolean-value = BIT
+
+ dev-float = dev-float-type WSP dev-float-value
+ dev-float-type = "DEVFLOAT" | "DevFloat"
+ dev-float-value = floatnumber ; single-precision normalized numbers in IEC 60559
+
+ dev-short = dev-short-type WSP dev-short-value
+ dev-short-type = "DEVSHORT" | "DevShort"
+ dev-short-value = ["-"] DIGIT* ; In the range of "(2^15) - 1" to "-(2^15)".
```

Comment:
Should the range be lower-upper limit, e.g. "In the range of -(2^15) to (2^15)-1"